extensions/Worker.Extensions.EventHubs/src/EventHubExtensionStartup.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Extensions.Azure;
+
+[assembly: WorkerExtensionStartup(typeof(EventHubExtensionStartup))]
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    public class EventHubExtensionStartup : WorkerExtensionStartup
+    {
+        public override void Configure(IFunctionsWorkerApplicationBuilder applicationBuilder)
+        {
+            if (applicationBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(applicationBuilder));
+            }
+
+            applicationBuilder.Services.AddAzureClientsCore(); // Adds AzureComponentFactory
```

Comment:
Ok will remove from here and SB.

Reply:
In that case, should the entire type be removed since this was the only configuration happening?

Reply:
That's a good question, we originally had the startup because we had to register the converter, then later added AddAzureClientsCore for MI - so without the need for either of those, I think it's safe to remove the type but probably best to test E2E and confirm it still works

Reply:
Looks like it is required:

System.TypeLoadException: Could not load type 'Microsoft.Azure.Functions.Worker.EventHubExtensionStartup' from assembly 'Microsoft.Azure.Functions.Worker.Extensions.EventHubs, Version=5.4.0.0, Culture=neutral, PublicKeyToken=551316b6919f366c'.
   at Microsoft.Azure.Functions.Worker.WorkerExtensionStartupCodeExecutor.Configure(IFunctionsWorkerApplicationBuilder applicationBuilder)
   at Microsoft.Extensions.DependencyInjection.ServiceCollectionExtensions.RunExtensionStartupCode(IFunctionsWorkerApplicationBuilder builder) in C:\Users\jolov\src\r\azure-functions-dotnet-worker\src\DotNetWorker.Core\Hosting\ServiceCollectionExtensions.cs:line 144
   at Microsoft.Extensions.DependencyInjection.ServiceCollectionExtensions.AddFunctionsWorkerCore(IServiceCollection services, Action`1 configure) in C:\Users\jolov\src\r\azure-functions-dotnet-worker\src\DotNetWorker.Core\Hosting\ServiceCollectionExtensions.cs:line 98
   at Microsoft.Extensions.DependencyInjection.ServiceCollectionExtensions.AddFunctionsWorkerDefaults(IServiceCollection services, Action`1 configure) in C:\Users\jolov\src\r\azure-functions-dotnet-worker\src\DotNetWorker\Hosting\ServiceCollectionExtensions.cs:line 39
   at Microsoft.Extensions.Hosting.WorkerHostBuilderExtensions.<>c__DisplayClass5_0.<ConfigureFunctionsWorkerDefaults>b__2(HostBuilderContext context, IServiceCollection services) in C:\Users\jolov\src\r\azure-functions-dotnet-worker\src\DotNetWorker\Hosting\WorkerHostBuilderExtensions.cs:line 166
   at Microsoft.Extensions.Hosting.HostBuilder.CreateServiceProvider()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
   at Program.<Main>$(String[] args) in C:\Users\jolov\RiderProjects\Company.FunctionApp7\Company.FunctionApp7\Program.cs:line 3

Reply:
Actually, nevermind. After I cleaned all the local nuget packages and rebuilt all dependencies it worked. I guess there was something that cached that attribute locally.